onnxruntime/contrib_ops/cuda/bert/fast_gelu_impl.cu

```diff
@@ -0,0 +1,120 @@
+/*
+ The implementation of this file is based on gelu plugin in TensorRT demo:
+ https://github.com/NVIDIA/TensorRT/tree/release/5.1/demo/BERT/
+ 
+Copyright 2019 NVIDIA Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Modifications: Add (bias) before Gelu is merged into this op to get better performance.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cuda_common.h"
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "core/providers/cuda/shared_inc/cuda_call.h"
+#include <cuda_fp16.h>
+#include <cublas_v2.h>
+#include "fast_gelu_impl.h"
+
+using namespace onnxruntime::cuda;
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+// constants for approximating the normal cdf
+constexpr float A = 0.5;
+
+constexpr float B = 0.7978845608028654;  // sqrt(2.0/M_PI)
+
+constexpr float C = 0.035677408136300125;  // 0.044715 * sqrt(2.0/M_PI)
+
+__device__ inline float Tanh(const float& x) {
+  return tanhf(x);
+}
+
+__device__ inline half Tanh(const half& x) {
+  const float tmp = tanhf(__half2float(x));
+  return __float2half(tmp);
+}
+
+__device__ inline half2 Tanh(const half2& x) {
+  // at the moment, there is no half2 tanh builtin
+  float2 tmp = (__half22float2(x));
+  tmp.x = tanhf(tmp.x);
+  tmp.y = tanhf(tmp.y);
+  return __float22half2_rn(tmp);
+}
```

Comment:
please reuse the code from common.cuh, extend it for half2.
[https://github.com/microsoft/onnxruntime/blob/6651d2f66259fc4723fc3855c9d7eb13b1af59e8/onnxruntime/core/providers/cuda/cu_inc/common.cuh#L146-L155](https://github.com/microsoft/onnxruntime/blob/6651d2f66259fc4723fc3855c9d7eb13b1af59e8/onnxruntime/core/providers/cuda/cu_inc/common.cuh#L146-L155)

Reply:
done